geoportailv3/static/js/authentication/authenticationdirective.js

```diff
@@ -0,0 +1,227 @@
+   * @private
+   */
+  this.logoutUrl_ = logoutUrl;
+
+  /**
+   * @type {Object}
+   * @private
+   */
+  this['credentials'] = {
+    'login' : null,
+    'password' : null
+  };
+
+  /**
+   * @type {angular.$http}
+   * @private
+   */
+  this.http_ = $http;
+
+  /**
+   * @type {string}
+   */
+  this['login'] = null;
+
+
+  /**
+   * @type {string}
+   */
+  this['email'] = null;
+
+
+  /**
+   * @type {string}
+   */
+  this['role'] = null;
+
+  /**
+   * @type {number}
+   */
+  this['role_id'] = 0;
+
+  /**
+   * @type {string}
+   */
+  this['name'] = null;
+
+  /**
+   * @type {boolean}
+   */
+  this['isError'] = false;
+
+  this.getUserInfo();
+};
+
+
+/**
+ * @param {Object} credentials Credentials.
+ * @export
+ */
+app.AuthenticationController.prototype.authenticate = function(credentials) {
+
+  var that = this;
+  var req = $.param({
+    'login': credentials['login'],
+    'password': credentials['password']
+  });
+  var config = {
+    headers: {'Content-Type': 'application/x-www-form-urlencoded'}
+  };
+  this.http_.post(this.loginUrl_, req, config).success(
+      function(data, status, headers, config) {
+        if (status == 200) {
+          that.getUserInfo();
+          that['isError'] = false;
+        }else {
+          that['isError'] = true;
+        }
+      }). error(function(data, status, headers, config) {
+    that['isError'] = true;
+  });
+};
+
+
+/**
+ * @export
+ */
+app.AuthenticationController.prototype.logout = function() {
+  var that = this;
+  var req = {};
+  var config = {
+    headers: {'Content-Type': 'application/x-www-form-urlencoded'}
+  };
+  this.http_.post(this.logoutUrl_, req, config).success(
+      function(data, status, headers, config) {
+        if (status == 200) {
+          that.getUserInfo();
+          that['isError'] = false;
+        }else {
+          that.getUserInfo();
+          that['isError'] = true;
+        }
+      }). error(function(data, status, headers, config) {
+    that.getUserInfo();
+    that['isError'] = true;
+  });
+};
+
+
+/**
+ * @export
+ */
+app.AuthenticationController.prototype.getUserInfo = function() {
+  var that = this;
+  var req = {};
+  var config = {
+    headers: {'Content-Type': 'application/x-www-form-urlencoded'}
+  };
+
+  this.http_.post(this.getuserinfoUrl_, req, config).success(
+      function(data, status, headers, config) {
+        if (status == 200) {
+          that.setUserInfo(
+              data['login'],
+              data['role'],
+              data['role_id'],
+              data['mail'],
+              data['sn']
+          );
+        }else {
+          that.setUserInfo(null, null, null, null, null);
+        }
+
+      }).error(function(data, status, headers, config) {
+    that.setUserInfo(null, null, null, null, null);
+  });
+};
+
+
+/**
+ * @return {boolean}
+ * @export
+ */
+app.AuthenticationController.prototype.isAuthenticated = function() {
+  if (!goog.isNull(this['login']) && this['login'].length > 0) {
```

Comment:
For the record, I think https://github.com/Geoportail-Luxembourg/geoportailv3/issues/275 is related to that line. It looks like there are cases when `this['login']` is `undefined` instead of `null`.